Add off hand support for Armor stands

Postremus opened the pull request:
Add equipment sounds for armorstands. They are played to the player who equipped the armor stand. Currently no sound is played because of #519.

Implement EntityEquipment. This allows the armorstand to use the equipmentmonitor.

Save the off hand item of the livingentity. Use the tags HandItems, ArmorItems, HandDropChances, ArmorDropChances for this. Keep the old Equipment and DropChances tags for backwards compatibility.

Add off hand support for armor stands.

closes #471

Postremus committed:
Add off_hand (shield) support for armorstands
also add equipment sounds
closes #471

Postremus committed:
Implement ArmorStand.getEquipment()

Postremus committed:
Fix some imports
some javadocs
fix implementation of setArmorContents

Postremus committed:
Implement drop chances

Postremus committed:
Store offhand equipment
implemented the HandItems, ArmorItems

Postremus committed:
Actually send the of hand item on armor stand spawn

Postremus committed:
revert changes in start.sh

aramperes commented on src/main/java/net/glowstone/entity/objects/GlowArmorStand.java:
```diff
@@ -1,13 +1,20 @@
 package net.glowstone.entity.objects;
 
+
```
Whitespace! ;)

aramperes commented on src/main/java/net/glowstone/io/entity/LivingEntityStore.java:
```diff
@@ -148,6 +130,68 @@ public void load(T entity, CompoundTag compound) {
+        // Deprecated since 15w31a, left here for compatibilty for now
+        if (compound.isList("Equipment", TagType.COMPOUND)) {
+            List<CompoundTag> list = compound.getCompoundList("Equipment");
+            if (list.size() == 5) {
```
Would it be better to allow for a variable size of this list? For example, having a list with only 1 element would only set the "item in hand" slot, and so forth.

Postremus replied:
I can't see a Situation where:
- A world is saved with an incomplete list of "Equipment", "HandItems" or "ArmorItems"
- and the client still working

Just tried this situation out. Client is just black, and I can't move.

Since it is not safe to have incomplete lists in vanilla, I don't see how useful handling loading of them is going to be for us.

Postremus replied:
Okay, more testing revealed, that paper has no problem with incomplete *Items lists.

aramperes replied:
The only situation I could see this happening is when using data-tags in commands (/summon, for example): it would be possible for a command to not have all the items in the list, and the Entity Store would have to handle that.

Postremus committed:
fix imports
don't call equipmentmonitor.reset() twice

Postremus committed:
Don't save equipment in armorstandstore
we already do that properly in the livingentity

Also don't save the *Items and *Chances Tag for a player, since he can never have these tags

Postremus committed:
Handle *Items and *Drops Tags with variable sized lists
order of the lists musts still be correct, e.g. feet must be before leg

If no Item or dropChance is available, load an empty Itemstack or 1f respectively

aramperes commented:
@Postremus I don't have time to test this on my own system right now. Can you confirm that your changes work without major breakage? If not, I'll have a look myself tonight or during the weekend.

Postremus commented:
It seems to work without breaking anything.   

Executing:
`/summon armor_stand ~ ~ ~ {ShowArms:1,Invisible:1,HandItems:[{Count:1b,id:276s},{Count:1b,id:442s}], ArmorItems:[{Count:1b,id:313s}, {Count:1b,id:312s}, {Count:1b,id:311s}]}`  
Spawns the armorstand with 3 pieces of armor (foot, leg, chest), and 2 handitems, correctly.

No other entity uses an EntityEquipment right now in Glowstone.

aramperes commented:
@Postremus Alright, thank you for confirming this works :)

Perhaps adding EntityEquipment support for other entities can be done in a future PR. The scope of your PR is mostly about Armor Stands anyways.

aramperes commented:
Thank you again for your contribution!
